tool_collections/kraken/kraken/kraken.xml

```diff
@@ -9,7 +9,23 @@
     <command>
         <![CDATA[
         kraken --threads \${GALAXY_SLOTS:-1} @INPUT_DATABASE@ 
+        
+        #if isinstance( $input_sequences.datatype, $__app__.datatypes_registry.get_datatype_by_extension('fastq').__class__):
```

Comment:
You are only allowing  'fastq, fasta' so it's probably easier to use $input_sequences.ext == 'fastq'